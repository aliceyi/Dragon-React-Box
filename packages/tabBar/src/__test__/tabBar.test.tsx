import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import TabBar, { TabBarProps } from '../index'
// https://github.com/styled-components/jest-styled-components
import 'jest-styled-components'

const testTabBars: TabBarProps = {
    classNames: 'abc',
    testData: 'tabBar-test-has-data',
    defaultIndex: 1,
    tabs: [
        {
            icon: 'IconEdit',
            iconProps: {
                width: '1.5rem',
                height: '1.5rem',
            },
            text: '发现',
            path: '/a',
        },
        {
            icon: 'IconPlus',
            iconProps: {
                width: '1.5rem',
                height: '1.5rem',
            },
            text: '探索',
            path: '/b',
        },
        {
            icon: 'IconGuide',
            iconProps: {
                width: '1.5rem',
                height: '1.5rem',
            },
            text: '远征',
            path: '/c',
        },
    ],
    onClick: jest.fn(),
}

describe('test Tabs component', () => {
    it('should render the correct default tabBar', () => {
        render(<TabBar {...testTabBars} />)
        const element = screen.getByTestId('tabBar-test-has-data')
        expect(element).toBeInTheDocument()
        expect(element.tagName).toEqual('DIV')
        expect(element).toMatchSnapshot()
    })
    it('should click tabBar to have been called', () => {
        render(<TabBar {...testTabBars} />)
        const element = screen.getByText(/发现/i)
        expect(element).toBeInTheDocument()
        fireEvent.click(element)
        expect(testTabBars.onClick).toHaveBeenCalled()
        expect(element).toMatchSnapshot()
    })
    it('get a render total number of tabBar item', () => {
        render(<TabBar {...testTabBars} />)
        const element = screen.getAllByText(/探索/i)
        expect(element).toHaveLength(1)
        expect(element).toMatchSnapshot()
    })
    it('get active item index when click tab', () => {
        render(<TabBar {...testTabBars} />)
        const element = screen.getByTestId(`${testTabBars.testData}_2`)
        console.log(element)
        expect(element).toBeInTheDocument()
        fireEvent.click(element)
        expect(element.getAttribute('data-testid')).toEqual(`${testTabBars.testData}_2`)
    })
})
