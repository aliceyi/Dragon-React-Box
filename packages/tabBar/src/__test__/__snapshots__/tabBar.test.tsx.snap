// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tabs component get a render total number of tabBar item 1`] = `
Array [
  <div>
    探索
  </div>,
]
`;

exports[`test Tabs component should click tabBar to have been called 1`] = `
<div>
  发现
</div>
`;

exports[`test Tabs component should render the correct default tabBar 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  min-height: 100vh;
  -webkit-overflow-scrolling: touch;
  padding-bottom: 48px;
  padding-bottom: calc(48px + constant(safe-area-inset-bottom));
  padding-bottom: calc(48px + env(safe-area-inset-bottom));
}

.c1 {
  position: fixed;
  width: 100%;
  left: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: $size-xxs + $size-xxxs 0 $size-xxs;
  height: 48px;
  font-family: 'PingFangSC-Semibold','sans-serif-black','Helvetica Neue For Number','-apple-system',BlinkMacSystemFont,'Segoe UI',Roboto,'Hiragino Sans GB','Microsoft YaHei','Helvetica Neue',Helvetica,Arial;
  font-size: 0.625rem;
  line-height: 1;
  background-color: #ffffff;
  text-align: center;
}

.c2 {
  width: 33.3%;
  cursor: pointer;
  color: #b6b6b6;
}

.c3 {
  height: 24px;
}

<div
  class=""
  data-testid="tabBar-test-has-data"
>
  <section
    class="c0"
  />
  <section
    class="c1"
  >
    <div
      class="c2"
      data-testid="tabBar-test-has-data_0"
    >
      0
      <div
        class="c3"
      >
        <svg
          height="1.5rem"
          viewBox="0 0 19 19"
          width="1.5rem"
        >
          <path
            d="M7.787 9.41l2.501 2.5-3.31.75.809-3.25zm8.85-3.834l-5.505 5.571L8.555 8.57 14.06 3l2.578 2.576zm-3.409 5.067v3.722H4.136v-6.82h3.832L9.092 6.41H3v9.092h11.365V9.493l-1.137 1.15z"
            fill="currentColor"
            fill-rule="nonzero"
          />
        </svg>
      </div>
      <div>
        发现
      </div>
    </div>
    <div
      class="c2"
      data-testid="tabBar-test-has-data_1"
      style="color: rgb(34, 34, 34);"
    >
      1
      <div
        class="c3"
      >
        <svg
          height="1.5rem"
          viewBox="0 0 19 19"
          width="1.5rem"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-width="2"
          >
            <path
              d="M9.445 3.159V15.73M15.73 9.445H3.16"
            />
          </g>
        </svg>
      </div>
      <div>
        探索
      </div>
    </div>
    <div
      class="c2"
      data-testid="tabBar-test-has-data_2"
    >
      2
      <div
        class="c3"
      >
        <svg
          height="1.5rem"
          viewBox="0 0 48 48"
          width="1.5rem"
        >
          <path
            d="M40 4v40H8V12a8 8 0 018-8h24zm-4 4h-6v16l-6-2-6 2V8h-2a4 4 0 00-4 4v28h24V8zM26 8h-4v10.45l2-.666 2 .666V8z"
            fill="currentColor"
            fill-rule="nonzero"
          />
        </svg>
      </div>
      <div>
        远征
      </div>
    </div>
  </section>
</div>
`;
